src/utils/contexts/balances/provider.tsx

```diff
@@ -0,0 +1,64 @@
+import { ChainBalance, CurrencyExt } from '@interlay/interbtc-api';
+import { createContext, useEffect, useRef } from 'react';
+import { QueryClient, useQueryClient } from 'react-query';
+import { useSelector } from 'react-redux';
+
+import { StoreType } from '@/common/types/util.types';
+import { useSubstrateSecureState } from '@/lib/substrate';
+import { useGetCurrencies } from '@/utils/hooks/api/use-get-currencies';
+
+import { BALANCES_QUERY_KEY } from './hook';
+import { BalanceData } from './types';
+
+type UnsubscriptionsRef = Array<() => void> | null;
+
+const updateBalances = (data: ChainBalance, queryClient: QueryClient) =>
+  queryClient.setQueriesData<BalanceData>(BALANCES_QUERY_KEY, (oldData) => ({
+    ...oldData,
+    [data.currency.ticker]: data
+  }));
+
+const subscribe = (currency: CurrencyExt, address: string, queryClient: QueryClient) => {
+  const callback = (_: string, balance: ChainBalance) => updateBalances(balance, queryClient);
+  return window.bridge.tokens.subscribeToBalance(currency, address, callback);
+};
+
+const unsubscribeAll = (refs: UnsubscriptionsRef) => {
+  refs?.map((unsubscribe) => unsubscribe());
+  refs = null;
+};
+
+const BalanceStateContext = createContext(undefined);
+
+const BalancesProvider: React.FC = ({ children }): JSX.Element => {
```

Comment:
Actually, what is the usage of the React context good for here? Since there is no value being passed through the context, won't it be possible to just handle this with a simple wrapper component that will be wrapping the app and managing the subscriptions?

Reply:
Because to me it looks basically like provider that does not provide any value.

Reply:
moved.